content/post/2020-03-03-packageRank/index.Rmd

```diff
@@ -0,0 +1,854 @@
+  smpl.archive.histories)
+
+v.ct <- vapply(histories, nrow, integer(1L))
+
+p.data <- data.frame(package = sample.pkgs, do.call(rbind, p.data),
+  ver.ct = v.ct, stringsAsFactors = FALSE)
+
+row.names(p.data) <- NULL
+
+p.data$size.bias <- 100 * (p.data$count - p.data$size) / p.data$size
+p.data$version.bias <- 100 * (p.data$count - p.data$version) / p.data$version
+p.data$size.version.bias <- 100 * (p.data$count - p.data$size.version) /
+  p.data$size.version
+
+# p.data$popularity <- p.data$count
+p.data$popularity <- log10(p.data$size.version)
+p.data$versions <- log10(p.data$ver.ct)
+p.data$bias <- log10(p.data$size.version.bias)
+
+ols1 <- lm(bias ~ popularity + versions, data = p.data)
+ols2 <- lm(bias ~ popularity + versions + popularity * versions, data = p.data)
+
+# car::avPlots(ols1, layout = c(1, 1), grid = FALSE)
+# car::avPlots(ols2, layout = c(1, 1), grid = FALSE)
+# sel <- names(coef(ols2)[-1])
+# f.test <- car::linearHypothesis(ols2, sel, coef(ols2)[sel])
+```
+
+```{r ols_summary, echo = FALSE}
+summary(ols2)
+```
+
+******
+
+# `conclusions`
+
+This post introduces some of the functions and features of [`'packageRank'`](https://cran.r-project.org/package=packageRank). The aim of the package is to put package download counts into context using visualization and rank percentiles. The post also describes a systematic, positive bias that affects download counts and offers some ideas about how to minimize its effect.
+
+The package is a work-in-progress. Suggestions, feature requests and problems can be submitted to the package's GitHub [issues](https://www.github.com/lindbrook/packageRank/issues/). Insights about "small" downloads would be particularly welcome.
+
+I conclude with two final bits of data. On February 29, 2020 (a leap day and a Sunday), there was a noticeable decline in traffic to CRAN. Barring some technical issue with [RStudio's logs](http://cran-logs.rstudio.com/), there wasn't anything in the real world that might explain the drop. In the R world however, the day was notable for the release of the source code for R version 3.6.3.
+
+docker, aws
+
+```{r cran_pkgs, fig.align = "center", hugoopts = list(title = "Figure 19 Effect of the 2020 Leap Day on Package Downloads")}
+plot(cranDownloads(from = "2020-02-25", to = "2020-03-04"),
+  r.version = TRUE)
+```
+
+<br/>
+
+```{r cran_r, fig.align = "center", hugoopts = list(title = "Figure 20 Effect of 2020 Leap Day on R Downloads")}
+plot(cranDownloads("R", from = "2020-02-25", to = "2020-03-04"),
+  r.version = TRUE)
+```
+
+<br/>
+
+I've noticed that website maintenance (renaming files and folders, etc.) on CRAN sometimes coincides with software updates. So my working hypothesis, which I've yet to explore, is that the scripts used to do automated downloading may have been "broken" by changes on CRAN. If so, this might actually be a better benchmark estimate of the "real" level of interest in R and its packages.
+
```

Comment:
Maybe you could add some call to action e.g. remind where the issue tracker of your package is? For folks interesting in discussions (although they might also use the comments field of the blog)

Reply:
I cover this a bit in the second paragraph of the conclusion. You think I should expand on it?

Reply:
Yes, the more explicit, the better, and really end with it, for folks who might skip parts but read the end.

Reply:
Dropped the leap day ending.